### ISSUE TITLE
How to evaluate the test datasets and get the best F1 score?  

### ISSUE THREAD

**Issue by zengjie617789** [USC-InfoLab/rddc2020#7]
I implement the code successfully and i can inference my datasets. but i donot know how to evaluate the test1 and test2 datasets which are lack of annotations file. Only we get the test results throught uploading the evaluated result to official websites online?
After i check the code, i think the way to get the f1 score is by running this command below:
`python test.py --weights weights/IMSC/last_95_640_16.pt --data data/road.yaml`
I will be appreciated if any help  is offered.
thank you in advance.

**Comment by GMN23362** [comment]
> I implement the code successfully and i can inference my datasets. but i donot know how to evaluate the test1 and test2 datasets which are lack of annotations file. Only we get the test results throught uploading the evaluated result to official websites online? After i check the code, i think the way to get the f1 score is by running this comman below: `python test.py --weights weights/IMSC/last_95_640_16.pt --data data/road.yaml` I will be appreciated if any help is offered. thank you in advance.

What is the address of the official websites? Hope to see your reply. Thank you very much!

**Comment by zengjie617789** [comment]
I think it is this: https://rdd2020.sekilab.global/

**Comment by GMN23362** [comment]
> I think it is this: https://rdd2020.sekilab.global/

Thanks for your reply! Have you solved the problem eventually? Do you still remember your F1-score?

**Comment by zengjie617789** [comment]
I split the Japan and Crech datasets into train and test which all have labels. the model is using YOLOX；
I got F1-score by calculating  the acc and recall on test data、 India and custom datasets；
the results are below:
![image](https://user-images.githubusercontent.com/25050291/163746713-aed131fd-fb6d-4e9f-8d9c-d599a42e24e5.png)

**Comment by GMN23362** [comment]
> I split the Japan and Crech datasets into train and test which all have labels. the model is using YOLOX； I got F1-score by calculating the acc and recall on test data、 India and custom datasets； the results are below: ![image](https://user-images.githubusercontent.com/25050291/163746713-aed131fd-fb6d-4e9f-8d9c-d599a42e24e5.png)

Thanks for your answer! I'd try it again.

**Comment by GMN23362** [comment]
> I split the Japan and Crech datasets into train and test which all have labels. the model is using YOLOX； I got F1-score by calculating the acc and recall on test data、 India and custom datasets； the results are below: ![image](https://user-images.githubusercontent.com/25050291/163746713-aed131fd-fb6d-4e9f-8d9c-d599a42e24e5.png)

Hello! May I ask your hyper-parameters? Have you changed default hyper-parameters? The default batch size is 64, but due to  my GPU limit, I changed the batch size to 8, which result in a f1-score of 0.46(really poor!). Hope to see your answer! Thank you very much!

**Comment by zengjie617789** [comment]
I didnot change any hyper-parameters in the config, BTW, the model i used is yolox instead of yolov5.

**Comment by GMN23362** [comment]
> I didnot change any hyper-parameters in the config, BTW, the model i used is yolox instead of yolov5.

Thanks for your reply. I used your approach, but the parameter '133' doesn't equals to the number of test images which is '4230'. Do you know what does '133' present for? Any help will be so appreciated!
![image](https://user-images.githubusercontent.com/74642115/164966960-6fc7dfea-3564-498d-a817-559e66d65814.png)

**Comment by zengjie617789** [comment]
@GMN23362  133 multiply batch size 32 equals the total num of imgs.
I received the ask in email while i canot see it here.

**Comment by GMN23362** [comment]
> @GMN23362 133 multiply batch size 32 equals the total num of imgs. I received the ask in email while i canot see it here.

Thank for your answer, I'll check it then.

**Comment by JIRRRR77** [comment]
![2022-07-09 08-59-29](https://user-images.githubusercontent.com/65073931/178085768-213ff1d6-3501-4a7f-ab29-8fa957da0088.png)
May i ask what is going on here when i run test.py? i did not change anythings but there is only 1 image scanned? labels.cache (0 found, 0 missing, 1 empty, 0 duplicate, for 1 images) and No labels found in datasets/road2020/train/Czech/labels/. 
and the P, R, mAP is 0.

**Comment by zengjie617789** [comment]
you should download the entire datasets and place in on certain path.

**Comment by MHassanlatif** [comment]
>![2022-07-09 08-59-29](https://user-images.githubusercontent.com/65073931/178085768-213ff1d6-3501-4a7f-ab29-8fa957da0088.png) May I ask what is going on here when I run test.py? I did not change anything but there is only 1 image scanned. labels.cache (0 found, 0 missing, 1 empty, 0 duplicate, for 1 images) and No labels found in datasets/road2020/train/Czech/labels/. and the P, R, and mAP are 0.

Did you get some solution to your problem? I am facing the same problem **P, R, and mAP are 0.**

**Comment by MHassanlatif** [comment]
> you should download the entire datasets and place in on the certain path.

I place my datasets files **(Japan, India, Czech)** file inside the **(datasets/road2020/) directory** but when I try to run the code using the data get by **road.yaml** file it shows 1 image scanned and no labels for Czech also when open the **(Val.txt file)** it has only one entry for one image. kindly guide me regarding this issue.
